src/InstrumentationEngine.Lib/ConfigurationSource.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "InstrumentationMethodSetting.h"
+
+namespace MicrosoftInstrumentationEngine
+{
+    class CConfigurationSource :
+        public CModuleRefCount,
+        public IUnknown
+    {
+    private:
+        CComBSTR m_bstrPath;
+        std::vector<ATL::CComPtr<IInstrumentationMethodSetting>> m_settings;
```

Comment:
Will we merge the two concepts together in the future? ie. parse the m_bstrPath into IInstrumentationMethodSettings? It feels disjoint that the constructor takes in the xml path but requires AddSetting calls to populate the vector.

Reply:
m_bstrPath is the path to the configuration file that contains the IM information. m_settings is effectively a key-value pair of settings that will be passed to the IM when it is initialized; such settings are the ones that each IM is currently setting via environment variables (e.g. MicrosoftProductionDiagnostics_RuntimeFolder in the case of the Production Breakpoints IM). This is needed because if you are attaching, you likely didn't have an opportunity to set environment variables on the process before it started.

Reply:
I am confused by the code in ProfilerManager that creates a new ConfigurationSource but calls EnumSettings, which would always return an empty vector.